File: removeElements.java
package edu.ucsc;

public class removeElements {
    public ListNode removeElements(ListNode head, int val) {
        ListNode cur = head;
        ListNode fakeHead = new ListNode(-1);
        fakeHead.next = head;
        ListNode prev = fakeHead;
        while (cur != null) {
            if (cur.val == val) {
                prev.next = cur.next;
            }else {
                prev = prev.next;
            }
            cur = cur.next;
        }
        return fakeHead.next;
    }
}

File: Python/LinkedList/removeElements.py
#! /Users/xiaotongli/anaconda3/bin/python
# -*- coding: utf-8 -*-
# @Time    : 10/6/18 10:45 AM
# @Author  : Xiaotong Li
# @School  : University of California, Santa Cruz
# @FileName: removeElements.py
# @Software: PyCharm


# Definition for singly-linked list.
class ListNode:
    def __init__(self, x):
        self.val = x
        self.next = None


class Solution:
    def removeElements(self, head, val):
        """
        :type head: ListNode
        :type val: int
        :rtype: ListNode
        """
        if not head:
            return head

        cur = head
        cur_next = cur.next

        while cur_next:
            if cur_next.val == val:
                cur_next = cur_next.next
                cur.next = cur_next
            else:
                cur = cur_next
                cur_next = cur_next.next

        if head.val == val:
            head = head.next

        return head
